ApiReview/API.Polly.Core/NoDocs/Polly/ResilienceStrategyBuilder.cs

```diff
@@ -0,0 +1,9 @@
+// Assembly 'Polly.Core'
+
+namespace Polly;
+
+public sealed class ResilienceStrategyBuilder : ResilienceStrategyBuilderBase
```

Comment:
As I've stated in [this issue](https://github.com/App-vNext/Polly/issues/1324) I think the  `ResilienceStrategyBuilder` name is a bit misleading. Since it is not meant to be used for building a single strategy rather than composing multiple strategies.

I think renaming it to `ResiliencePipelineBuilder` or similar would better express the intent. 

V8 introduces a brand new API with a couple of new concepts. IMHO they should crystal clearly communicate what component should be used when.

Reply:
My personal preference is to stick with `ResilienceStrategy` everywhere, as the "strategy" can represent both simple and complex behavior consisting of retries, timeouts, etc.

Reply:
I second Peter's point - I also think that pipeline != strategy and we'd better have the split upfront

Reply:
Is it ok if it returns `ResilienceStrategy` although it is named `ResiliencePipelineBuilder`?

Introducing new type such as `ResiliencePipeline` is something I'd like to avoid, otherwise I am fine with @PeterCsalaHbo suggestion.

Reply:
`CompositeStrategyBuilder`

Reply:
@martintmk Sorry I was on vacation in the past few weeks. I like the new name. Thanks for the renaming! 👍